### ISSUE TITLE
Installation assistant fails to remove wazuh component upon lack of space error

### ISSUE THREAD

**Issue by Rebits** [wazuh/wazuh-installation-assistant#5]
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.8.0-beta5 | Wazuh installation assistant | Install | Ubuntu 22 |

### Description

It has been detected that in case of failing the installation of the wazuh-dashboard due to lack of space in the system, the wazuh component previously installed are not removed correctly 

### Steps to reproduce
- Deploy an instance with not enough disk space for AIO installation
- Download assistance and install all component in the instance: `curl -sO https://packages-dev.wazuh.com/4.8/wazuh-install.sh && sudo bash ./wazuh-install.sh -a`
- After failure check that wazuh-manager and wazuh-indexer are installed

```
root@ip-172-31-44-9:/home/ubuntu# apt list --installed | grep wazuh-manager
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

wazuh-manager/now 4.8.0-1 amd64 [installed,local]
root@ip-172-31-44-9:/home/ubuntu# apt list --installed | grep wazuh-indexer

wazuh-indexer/now 4.8.0-1 amd64 [installed,local]

```

- Check that the uninstall process removes configuration files

```
root@ip-172-31-44-9:/home/ubuntu# ls /var/ossec
ls: cannot access '/var/ossec': No such file or directory


root@ip-172-31-44-9:/home/ubuntu# ls /etc | grep wazuh

```

### Evidence

<details>

<summary> Installation log </summary>

- First installation
```
root@ip-172-31-44-9:/home/ubuntu# curl -sO https://packages-dev.wazuh.com/4.8/wazuh-install.sh && sudo bash ./wazuh-install.sh -a
12/04/2024 08:46:28 INFO: Starting Wazuh installation assistant. Wazuh version: 4.8.0
12/04/2024 08:46:28 INFO: Verbose logging redirected to /var/log/wazuh-install.log
12/04/2024 08:46:38 INFO: Verifying that your system meets the recommended minimum hardware requirements.
12/04/2024 08:46:38 INFO: Wazuh web interface port will be 443.
12/04/2024 08:46:44 INFO: --- Dependencies ----
12/04/2024 08:46:44 INFO: Installing apt-transport-https.
12/04/2024 08:46:54 INFO: Wazuh development repository added.
12/04/2024 08:46:54 INFO: --- Configuration files ---
12/04/2024 08:46:54 INFO: Generating configuration files.
12/04/2024 08:46:54 INFO: Generating the root certificate.
12/04/2024 08:46:54 INFO: Generating Admin certificates.
12/04/2024 08:46:55 INFO: Generating Wazuh indexer certificates.
12/04/2024 08:46:56 INFO: Generating Filebeat certificates.
12/04/2024 08:46:56 INFO: Generating Wazuh dashboard certificates.
12/04/2024 08:46:57 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
12/04/2024 08:46:57 INFO: --- Wazuh indexer ---
12/04/2024 08:46:57 INFO: Starting Wazuh indexer installation.
12/04/2024 08:48:36 INFO: Wazuh indexer installation finished.
12/04/2024 08:48:36 INFO: Wazuh indexer post-install configuration finished.
12/04/2024 08:48:36 INFO: Starting service wazuh-indexer.
12/04/2024 08:49:01 INFO: wazuh-indexer service started.
12/04/2024 08:49:01 INFO: Initializing Wazuh indexer cluster security settings.
12/04/2024 08:49:12 INFO: Wazuh indexer cluster security configuration initialized.
12/04/2024 08:49:12 INFO: Wazuh indexer cluster initialized.
12/04/2024 08:49:12 INFO: --- Wazuh server ---
12/04/2024 08:49:12 INFO: Starting the Wazuh manager installation.
12/04/2024 08:50:46 INFO: Wazuh manager installation finished.
12/04/2024 08:50:46 INFO: Wazuh manager vulnerability detection configuration finished.
12/04/2024 08:50:46 INFO: Starting service wazuh-manager.
12/04/2024 08:51:07 INFO: wazuh-manager service started.
12/04/2024 08:51:07 INFO: Starting Filebeat installation.
12/04/2024 08:51:22 INFO: Filebeat installation finished.
12/04/2024 08:51:24 INFO: Filebeat post-install configuration finished.
12/04/2024 08:51:24 INFO: Starting service filebeat.
12/04/2024 08:51:26 INFO: filebeat service started.
12/04/2024 08:51:26 INFO: --- Wazuh dashboard ---
12/04/2024 08:51:26 INFO: Starting Wazuh dashboard installation.
12/04/2024 08:51:36 ERROR: Wazuh dashboard installation failed.
12/04/2024 08:51:36 INFO: --- Removing existing Wazuh installation ---
12/04/2024 08:51:39 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```
</details>

<details>

<summary> <code>/var/log/wazuh-install.log</code></summary>

```
root@ip-172-31-44-9:/home/ubuntu# cat /var/log/wazuh-install.log^C
root@ip-172-31-44-9:/home/ubuntu# cat /var/log/wazuh-install.log
12/04/2024 08:46:28 INFO: Starting Wazuh installation assistant. Wazuh version: 4.8.0
12/04/2024 08:46:28 INFO: Verbose logging redirected to /var/log/wazuh-install.log
Hit:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease
Reading package lists...
12/04/2024 08:46:38 INFO: Verifying that your system meets the recommended minimum hardware requirements.
12/04/2024 08:46:38 INFO: Wazuh web interface port will be 443.
Hit:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease
Reading package lists...
12/04/2024 08:46:44 INFO: --- Dependencies ----
12/04/2024 08:46:44 INFO: Installing apt-transport-https.
Reading package lists... Building dependency tree... Reading state information... The following NEW packages will be installed: apt-transport-https 0 upgraded, 1 newly installed, 0 to remove and 43 not upgraded. Need to get 1,510 B of archives. After this operation, 170 kB of additional disk space will be used. Get:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-updates/ NEEDRESTART-VER: 3.5 NEEDRESTART-KCUR: 6.5.0-1014-aws NEEDRESTART-KEXP: 6.5.0-1014-aws NEEDRESTART-KSTA: 1sly unselected package apt-transport-https.
gpg: keyring '/usr/share/keyrings/wazuh.gpg' created
gpg: directory '/root/.gnupg' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key 96B3EE5F29111145: public key "Wazuh.com (Wazuh Signing Key) <support@wazuh.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages-dev.wazuh.com/pre-release/apt/ unstable main
Hit:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
Hit:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:4 http://us-east-1.ec2.archive.ubuntu.com/ubuntu jammy-backports InRelease
Get:5 https://packages-dev.wazuh.com/pre-release/apt unstable InRelease [17.3 kB]
Get:6 https://packages-dev.wazuh.com/pre-release/apt unstable/main amd64 Packages [37.1 kB]
Fetched 54.4 kB in 1s (41.8 kB/s)
Reading package lists...
12/04/2024 08:46:54 INFO: Wazuh development repository added.
12/04/2024 08:46:54 INFO: --- Configuration files ---
12/04/2024 08:46:54 INFO: Generating configuration files.
12/04/2024 08:46:54 INFO: Generating the root certificate.
12/04/2024 08:46:54 INFO: Generating Admin certificates.
12/04/2024 08:46:55 INFO: Generating Wazuh indexer certificates.
12/04/2024 08:46:56 INFO: Generating Filebeat certificates.
12/04/2024 08:46:56 INFO: Generating Wazuh dashboard certificates.
12/04/2024 08:46:57 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
12/04/2024 08:46:57 INFO: --- Wazuh indexer ---
12/04/2024 08:46:57 INFO: Starting Wazuh indexer installation.
Reading package lists... Building dependency tree... Reading state information... The following NEW packages will be installed: wazuh-indexer 0 upgraded, 1 newly installed, 0 to remove and 43 not upgraded. Need to get 757 MB of archives. After this operation, 1,050 MB of additional disk space will be used. Get:1 https://packages-dev.wazuh.com/pre-release/apt unstable/main amd64 NEEDRESTART-VER: 3.5 NEEDRESTART-KCUR: 6.5.0-1014-aws NEEDRESTART-KEXP: 6.5.0-1014-aws NEEDRESTART-KSTA: 1kage wazuh-indexer.
12/04/2024 08:48:36 INFO: Wazuh indexer installation finished.
12/04/2024 08:48:36 INFO: Wazuh indexer post-install configuration finished.
12/04/2024 08:48:36 INFO: Starting service wazuh-indexer.
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-indexer.service → /lib/systemd/system/wazuh-indexer.service.
12/04/2024 08:49:01 INFO: wazuh-indexer service started.
12/04/2024 08:49:01 INFO: Initializing Wazuh indexer cluster security settings.
**************************************************************************
** This tool will be deprecated in the next major release of OpenSearch **
** https://github.com/opensearch-project/security/issues/1755           **
**************************************************************************
Security Admin v7
Will connect to 127.0.0.1:9200 ... done
Connected as "CN=admin,OU=Wazuh,O=Wazuh,L=California,C=US"
OpenSearch Version: 2.10.0
Contacting opensearch cluster 'opensearch' and wait for YELLOW clusterstate ...
Clustername: wazuh-cluster
Clusterstate: GREEN
Number of nodes: 1
Number of data nodes: 1
.opendistro_security index does not exists, attempt to create it ... done (0-all replicas)
Populate config from /etc/wazuh-indexer/opensearch-security/
Will update '/config' with /etc/wazuh-indexer/opensearch-security/config.yml 
   SUCC: Configuration for 'config' created or updated
Will update '/roles' with /etc/wazuh-indexer/opensearch-security/roles.yml 
   SUCC: Configuration for 'roles' created or updated
Will update '/rolesmapping' with /etc/wazuh-indexer/opensearch-security/roles_mapping.yml 
   SUCC: Configuration for 'rolesmapping' created or updated
Will update '/internalusers' with /etc/wazuh-indexer/opensearch-security/internal_users.yml 
   SUCC: Configuration for 'internalusers' created or updated
Will update '/actiongroups' with /etc/wazuh-indexer/opensearch-security/action_groups.yml 
   SUCC: Configuration for 'actiongroups' created or updated
Will update '/tenants' with /etc/wazuh-indexer/opensearch-security/tenants.yml 
   SUCC: Configuration for 'tenants' created or updated
Will update '/nodesdn' with /etc/wazuh-indexer/opensearch-security/nodes_dn.yml 
   SUCC: Configuration for 'nodesdn' created or updated
Will update '/whitelist' with /etc/wazuh-indexer/opensearch-security/whitelist.yml 
   SUCC: Configuration for 'whitelist' created or updated
Will update '/audit' with /etc/wazuh-indexer/opensearch-security/audit.yml 
   SUCC: Configuration for 'audit' created or updated
Will update '/allowlist' with /etc/wazuh-indexer/opensearch-security/allowlist.yml 
   SUCC: Configuration for 'allowlist' created or updated
SUCC: Expected 10 config types for node {"updated_config_types":["allowlist","tenants","rolesmapping","nodesdn","audit","roles","whitelist","internalusers","actiongroups","config"],"updated_config_size":10,"message":null} is 10 (["allowlist","tenants","rolesmapping","nodesdn","audit","roles","whitelist","internalusers","actiongroups","config"]) due to: null
Done with success
12/04/2024 08:49:12 INFO: Wazuh indexer cluster security configuration initialized.
12/04/2024 08:49:12 INFO: Wazuh indexer cluster initialized.
12/04/2024 08:49:12 INFO: --- Wazuh server ---
12/04/2024 08:49:12 INFO: Starting the Wazuh manager installation.
Reading package lists... Building dependency tree... Reading state information... Suggested packages: expect The following NEW packages will be installed: wazuh-manager 0 upgraded, 1 newly installed, 0 to remove and 43 not upgraded. Need to get 311 MB of archives. After this operation, 914 MB of additional disk space will be used. Get:1 https://packages-dev.wazuh.com/pre-releas NEEDRESTART-VER: 3.5 NEEDRESTART-KCUR: 6.5.0-1014-aws NEEDRESTART-KEXP: 6.5.0-1014-aws NEEDRESTART-KSTA: 1reviously unselected package wazuh-manager.
12/04/2024 08:50:46 INFO: Wazuh manager installation finished.
12/04/2024 08:50:46 INFO: Wazuh manager vulnerability detection configuration finished.
12/04/2024 08:50:46 INFO: Starting service wazuh-manager.
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-manager.service → /lib/systemd/system/wazuh-manager.service.
12/04/2024 08:51:07 INFO: wazuh-manager service started.
12/04/2024 08:51:07 INFO: Starting Filebeat installation.
Reading package lists... Building dependency tree... Reading state information... The following NEW packages will be installed: filebeat 0 upgraded, 1 newly installed, 0 to remove and 43 not upgraded. Need to get 22.1 MB of archives. After this operation, 73.6 MB of additional disk space will be used. Get:1 https://packages-dev.wazuh.com/pre-release/apt unstable/main amd64 file NEEDRESTART-VER: 3.5 NEEDRESTART-KCUR: 6.5.0-1014-aws NEEDRESTART-KEXP: 6.5.0-1014-aws NEEDRESTART-KSTA: 1eat.
12/04/2024 08:51:22 INFO: Filebeat installation finished.
wazuh/
wazuh/_meta/
wazuh/_meta/docs.asciidoc
wazuh/_meta/fields.yml
wazuh/_meta/config.yml
wazuh/alerts/
wazuh/alerts/config/
wazuh/alerts/config/alerts.yml
wazuh/alerts/manifest.yml
wazuh/alerts/ingest/
wazuh/alerts/ingest/pipeline.json
wazuh/module.yml
wazuh/archives/
wazuh/archives/config/
wazuh/archives/config/archives.yml
wazuh/archives/manifest.yml
wazuh/archives/ingest/
wazuh/archives/ingest/pipeline.json
Created filebeat keystore
Successfully updated the keystore
Successfully updated the keystore
12/04/2024 08:51:24 INFO: Filebeat post-install configuration finished.
12/04/2024 08:51:24 INFO: Starting service filebeat.
Synchronizing state of filebeat.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable filebeat
Created symlink /etc/systemd/system/multi-user.target.wants/filebeat.service → /lib/systemd/system/filebeat.service.
12/04/2024 08:51:26 INFO: filebeat service started.
12/04/2024 08:51:26 INFO: --- Wazuh dashboard ---
12/04/2024 08:51:26 INFO: Starting Wazuh dashboard installation.
Reading package lists... Building dependency tree... Reading state information... The following NEW packages will be installed: wazuh-dashboard 0 upgraded, 1 newly installed, 0 to remove and 43 not upgraded. Need to get 186 MB of archives. After this operation, 988 MB of additional disk space will be used. Get:1 https://packages-dev.wazuh.com/pre-release/apt unstable/main amd64 E: Sub-process /usr/bin/dpkg returned an error code (2)ut 'bolt' to '/var/lib/dpkg/status': No space left on devicetes
12/04/2024 08:51:36 ERROR: Wazuh dashboard installation failed.
12/04/2024 08:51:36 INFO: --- Removing existing Wazuh installation ---
12/04/2024 08:51:39 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
root@ip-172-31-44-9:/home/ubuntu# df -h
Filesystem       Size  Used Avail Use% Mounted on
/dev/root        7.6G  4.2G  3.4G  56% /
tmpfs            3.9G  516K  3.9G   1% /dev/shm
tmpfs            1.6G  872K  1.6G   1% /run
tmpfs            5.0M     0  5.0M   0% /run/lock
efivarfs         128K  3.6K  120K   3% /sys/firmware/efi/efivars
/dev/nvme0n1p15  105M  6.1M   99M   6% /boot/efi
tmpfs            784M  4.0K  784M   1% /run/user/1000
```
</details>

**Comment by c-bordon** [comment]
For this issue, we need  to investigate:

- Amount of storage required for each Wazuh core component for installation
- Amount of storage required for the operation of each Wazuh core component
- Identify all installation directories for each component

The installation assistant must be able to identify the storage required at each installation point since the user may have different disks mounted in the different directories. We must be able to perform the prior analysis, according to the type of installation (AIO or distributed) and by component.

**Comment by Enaraque** [comment]
This issue goes on hold due to higher priority tasks

**Comment by Enaraque** [comment]
# Update report

The main directories used in the installation of Wazuh components have been investigated. 

The affected directories are common to all. These are:
- `/usr/share/`
- `/etc/`
- `/var/cache/apt/archives/`
- `/var/lib/`

In addition, the Wazuh manager makes use of `/var/ossec/`.

Once we have figured out which directories each component uses, we have to estimate a used size of each one. To do this, we will download an AIO with both package managers (APT and RPM) and get the total size taken by each component in those directories.

## Next approach

Get the total size taken by each component in each directory. After obtaining the necessary information, we will start developing the code logic.

**Comment by Enaraque** [comment]
# Update report

To view the size of each directory, a Bash script has been created using the command with :
```
 sudo find / -name "*<WAZUH_COMPONENTS>*" -exec du -sh --block-size=1M {} \; | grep <DIRECTORY> | awk '{sum += $1} END {print sum}'
```
where 
```
COMPONENTS=("wazuh-manager" "wazuh-indexer" "wazuh-dashboard" "filebeat") 
DIRECTORIES=("/var/lib" "/etc" "/usr/share" "/var/ossec")`. 
```
This command is executed for each Wazuh component and each directory to determine the total size of the files generated for each component. This allows us to see the total size of the files generated for each Wazuh component. This has been done for both APT and RPM. The results are as follows:

<details><summary>Total size for APT</summary>

```console
root@ip-172-31-46-84:/home/ubuntu# bash check_space.sh

---------------- wazuh-manager ------------------------
wazuh-manager (ossec files) in /var/lib takes 12 MB
wazuh-manager (ossec files) in /etc takes 1 MB
wazuh-manager (ossec files) in /usr/share takes 0 MB
wazuh-manager (ossec files) in /var/ossec takes 6968 MB

---------------- wazuh-indexer ------------------------
wazuh-indexer in /var/lib takes 9 MB
wazuh-indexer in /etc takes 5 MB
wazuh-indexer in /usr/share takes 1030 MB
wazuh-indexer in /var/ossec takes 0 MB

---------------- wazuh-dashboard ------------------------
wazuh-dashboard in /var/lib takes 25 MB
wazuh-dashboard in /etc takes 6 MB
wazuh-dashboard in /usr/share takes 1118 MB
wazuh-dashboard in /var/ossec takes 0 MB

---------------- filebeat ------------------------
filebeat in /var/lib takes 7 MB
filebeat in /etc takes 4 MB
filebeat in /usr/share takes 143 MB
filebeat in /var/ossec takes 0 MB

---------------- Total size per directory in MB -----------------------------
Total space taken from /etc:    16
Total space taken from /usr/share:    2291
Total space taken from /var/lib:    53
Total space taken from /var/ossec:    6968
---------------- Total space taken up by the installation -----------------------------
Total space in MB: 9328
Total space in GB: 9
```

</details>

<details><summary>Total size for RPM</summary>

```console
[root@ip-172-31-47-54 ec2-user]# bash check_space.sh

---------------- wazuh-manager ------------------------
wazuh-manager (ossec files) in /var/lib takes 0 MB
wazuh-manager (ossec files) in /etc takes 1 MB
wazuh-manager (ossec files) in /usr/share takes 0 MB
wazuh-manager (ossec files) in /var/ossec takes 6995 MB

---------------- wazuh-indexer ------------------------
wazuh-indexer in /var/lib takes 3 MB
wazuh-indexer in /etc takes 5 MB
wazuh-indexer in /usr/share takes 1034 MB
wazuh-indexer in /var/ossec takes 0 MB

---------------- wazuh-dashboard ------------------------
wazuh-dashboard in /var/lib takes 0 MB
wazuh-dashboard in /etc takes 5 MB
wazuh-dashboard in /usr/share takes 1114 MB
wazuh-dashboard in /var/ossec takes 0 MB

---------------- filebeat ------------------------
filebeat in /var/lib takes 3 MB
filebeat in /etc takes 4 MB
filebeat in /usr/share takes 142 MB
filebeat in /var/ossec takes 0 MB

---------------- Total size per directory in MB -----------------------------
Total space taken from /etc:    15
Total space taken from /usr/share:    2290
Total space taken from /var/lib:    6
Total space taken from /var/ossec:    6995
---------------- Total space taken up by the installation -----------------------------
Total space in MB: 9306
Total space in GB: 9
```

</details>

## Conclusions

We can now provide an approximate value of the space that each directory will occupy after installation:

`/etc` -> 20MB
`/usr/share` -> 2350MB
`/var/lib` -> 65MB
`/var/ossec `-> 7100MB
This approximate value is intentionally set higher than the actual space occupied to allow for an acceptable margin of error.

**Comment by Enaraque** [comment]
# Update report

Part of the logic has been implemented to check that each Wazuh component has enough available space in each directory that will be used. This logic also includes a check to sum the required space if two directories are on the same partition. For example, if there are two directories with sizes 20 MB and 67 MB in `/var` and `/var/lib`, respectively, but both are located on the same partition `"dev-partition"` the total required space on that partition will be 97 MB. This ensures that the correct space is verified in each case.

Currently, this function is implemented for each individual component, but the idea is to extend it for the AIO (All-In-One) installation. This would check the total space required for all directories from each component, ensuring that there is enough space before starting the AIO installation, instead of checking during the installation of each Wazuh component.


## Tests

If we perform an AIO installation where there is not enough space for the complete installation (in this case, using an 8GB disk), we can see how the system checks for sufficient space in each component. When it reaches the installation of the Wazuh Manager, it verifies that there is not enough space and proceeds to remove what has already been installed without any issues.

```console
root@ip-172-31-35-199:/home/ubuntu# bash wazuh-install.sh -a
10/09/2024 16:08:41 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
10/09/2024 16:08:41 INFO: Verbose logging redirected to /var/log/wazuh-install.log
10/09/2024 16:09:05 INFO: Verifying that your system meets the recommended minimum hardware requirements.
10/09/2024 16:09:05 INFO: Wazuh web interface port will be 443.
10/09/2024 16:09:14 INFO: --- Dependencies ----
10/09/2024 16:09:14 INFO: Installing apt-transport-https.
10/09/2024 16:09:22 INFO: Installing debhelper.
10/09/2024 16:09:47 INFO: Wazuh repository added.
10/09/2024 16:09:47 INFO: --- Configuration files ---
10/09/2024 16:09:47 INFO: Generating configuration files.
10/09/2024 16:09:48 INFO: Generating the root certificate.
10/09/2024 16:09:48 INFO: Generating Admin certificates.
10/09/2024 16:09:48 INFO: Generating Wazuh indexer certificates.
10/09/2024 16:09:49 INFO: Generating Filebeat certificates.
10/09/2024 16:09:49 INFO: Generating Wazuh dashboard certificates.
10/09/2024 16:09:50 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
10/09/2024 16:09:50 INFO: --- Wazuh indexer ---
10/09/2024 16:09:50 INFO: Starting Wazuh indexer installation.
Partition: /dev/root, Required space: 1075, Available space: 5570
10/09/2024 16:10:28 INFO: Wazuh indexer installation finished.
10/09/2024 16:10:29 INFO: Wazuh indexer post-install configuration finished.
10/09/2024 16:10:29 INFO: Starting service wazuh-indexer.
10/09/2024 16:10:53 INFO: wazuh-indexer service started.
10/09/2024 16:10:53 INFO: Initializing Wazuh indexer cluster security settings.
10/09/2024 16:11:03 INFO: Wazuh indexer cluster security configuration initialized.
10/09/2024 16:11:03 INFO: Wazuh indexer cluster initialized.
10/09/2024 16:11:03 INFO: --- Wazuh server ---
10/09/2024 16:11:03 INFO: Starting the Wazuh manager installation.
Partition: /dev/root, Required space: 7105, Available space: 3728
10/09/2024 16:11:03 ERROR: There is not enough disk space in partition /dev/root to install wazuh-manager. Required: 7105MB, Available: 3728MB.
10/09/2024 16:11:03 INFO: --- Removing existing Wazuh installation ---
10/09/2024 16:11:03 INFO: Removing Wazuh indexer.
10/09/2024 16:11:14 INFO: Wazuh indexer removed.
10/09/2024 16:11:14 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
10/09/2024 16:11:16 ERROR: Wazuh installation failed.
10/09/2024 16:11:16 INFO: --- Removing existing Wazuh installation ---
10/09/2024 16:11:16 INFO: Wazuh GPG key not found in the system
10/09/2024 16:11:17 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
root@ip-172-31-35-199:/home/ubuntu# apt list --installed | grep wazuh-indexer

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
```

**Comment by Enaraque** [comment]
# Update report

Additionally, a check has been added so that during an AIO installation, the total available space is verified before the installation of any component.

```console
root@ip-172-31-35-199:/home/ubuntu# bash wazuh-install.sh -a
11/09/2024 09:17:51 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 09:17:51 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 09:18:00 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 09:18:00 INFO: Wazuh web interface port will be 443.
11/09/2024 09:18:12 INFO: Wazuh repository added.
11/09/2024 09:18:12 INFO: --- Configuration files ---
11/09/2024 09:18:12 INFO: Generating configuration files.
11/09/2024 09:18:13 INFO: Generating the root certificate.
11/09/2024 09:18:13 INFO: Generating Admin certificates.
11/09/2024 09:18:14 INFO: Generating Wazuh indexer certificates.
11/09/2024 09:18:14 INFO: Generating Filebeat certificates.
11/09/2024 09:18:15 INFO: Generating Wazuh dashboard certificates.
11/09/2024 09:18:16 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
11/09/2024 09:18:16 ERROR: There is not enough disk space in partition /dev/root to install AIO. Required: 9535MB, Available: 4133MB.
11/09/2024 09:18:16 INFO: --- Removing existing Wazuh installation ---
11/09/2024 09:18:17 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

**Comment by Enaraque** [comment]
# Update report

To make the installation process more efficient, the check for available disk space will be moved to just before installing any dependency or component. If we attempt to install the AIO without sufficient space, the output will now appear as follows: 

```console
[root@ip-172-31-35-162 ec2-user]# bash wazuh-install.sh -a
11/09/2024 16:41:41 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 16:41:41 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 16:41:41 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 16:41:41 INFO: Wazuh web interface port will be 443.
11/09/2024 16:41:41 ERROR: There is not enough disk space in partition /dev/nvme0n1p1 to install AIO. Required: 9535MB, Available: 6461MB.
11/09/2024 16:41:41 INFO: --- Removing existing Wazuh installation ---
11/09/2024 16:41:41 INFO: Wazuh GPG key not found in the system
11/09/2024 16:41:42 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

The installation has also been tested with different disk partitions. In this case, the disk was partitioned with a separate partition for `/usr/share`.

If we try to install an AIO with this partitioning, the result will be: 

```console
root@ip-172-31-35-199:/home/ubuntu# df -h
Filesystem       Size  Used Avail Use% Mounted on
/dev/root        7.6G  3.6G  4.1G  47% /
tmpfs            3.9G     0  3.9G   0% /dev/shm
tmpfs            1.6G  852K  1.6G   1% /run
tmpfs            5.0M     0  5.0M   0% /run/lock
/dev/nvme0n1p15  105M  6.1M   99M   6% /boot/efi
tmpfs            787M  4.0K  787M   1% /run/user/1000
/dev/nvme0n1p2   1.8G  249M  1.5G  15% /usr/share
root@ip-172-31-35-199:/home/ubuntu# bash wazuh-install.sh -a
11/09/2024 14:40:46 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 14:40:46 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 14:40:57 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 14:40:57 INFO: Wazuh web interface port will be 443.
11/09/2024 14:40:57 ERROR: There is not enough disk space in partition /dev/nvme0n1p2 to install AIO. Required: 2350MB, Available: 1479MB.
11/09/2024 14:40:57 INFO: --- Removing existing Wazuh installation ---
11/09/2024 14:40:57 INFO: Wazuh GPG key not found in the system
11/09/2024 14:40:57 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.  
```

This demonstrates the correct functionality of the space check, ensuring that each partition has the required space.


## Next approach

We are currently testing the behaviour when installing components separately.

**Comment by Enaraque** [comment]
# Update report

Tests have been conducted to verify the correct functioning. These tests were performed for both the distributed deployment (installing each component separately with the installation assistant) and for an AIO installation.

The tests had two parts: verifying that the installation proceeds correctly when there is sufficient space and verifying that the installation fails if there is insufficient space in a disk partition.

## Tests

### Installation with enough space

<details><summary>AIO installation</summary>

```console
$ bash wazuh-install.sh -a
11/09/2024 09:17:51 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 09:17:51 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 09:18:00 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 09:18:00 INFO: Wazuh web interface port will be 443.
11/09/2024 09:18:12 INFO: Wazuh repository added.
11/09/2024 09:18:12 INFO: --- Configuration files ---
11/09/2024 09:18:12 INFO: Generating configuration files.
11/09/2024 09:18:13 INFO: Generating the root certificate.
11/09/2024 09:18:13 INFO: Generating Admin certificates.
11/09/2024 09:18:14 INFO: Generating Wazuh indexer certificates.
11/09/2024 09:18:14 INFO: Generating Filebeat certificates.
11/09/2024 09:18:15 INFO: Generating Wazuh dashboard certificates.
11/09/2024 09:18:16 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
Partition: /dev/root, Required space: 9535, Available space: 4133
11/09/2024 09:18:16 ERROR: There is not enough disk space in partition /dev/root to install AIO. Required: 9535MB, Available: 4133MB.
11/09/2024 09:18:16 INFO: --- Removing existing Wazuh installation ---
11/09/2024 09:18:17 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

</details>

<details><summary>Wazuh Indexer</summary>

```console
$ bash wazuh-install.sh -wi node_indexer
11/09/2024 15:39:44 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 15:39:44 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 15:39:44 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 15:39:44 INFO: Wazuh repository added.
11/09/2024 15:39:45 INFO: --- Wazuh indexer ---
11/09/2024 15:39:45 INFO: Starting Wazuh indexer installation.
11/09/2024 15:40:35 INFO: Wazuh indexer installation finished.
11/09/2024 15:40:35 INFO: Wazuh indexer post-install configuration finished.
11/09/2024 15:40:35 INFO: Starting service wazuh-indexer.
11/09/2024 15:41:01 INFO: wazuh-indexer service started.
11/09/2024 15:41:01 INFO: Initializing Wazuh indexer cluster security settings.
11/09/2024 15:41:02 INFO: Wazuh indexer cluster initialized.
11/09/2024 15:41:02 INFO: Installation finished.

$ systemctl status wazuh-indexer
● wazuh-indexer.service - wazuh-indexer
     Loaded: loaded (/usr/lib/systemd/system/wazuh-indexer.service; enabled; preset: disabled)
     Active: active (running) since Wed 2024-09-11 15:41:01 UTC; 37s ago
       Docs: https://documentation.wazuh.com
   Main PID: 49637 (java)
      Tasks: 69 (limit: 9373)
     Memory: 1.3G
        CPU: 54.746s
     CGroup: /system.slice/wazuh-indexer.service
             └─49637 /usr/share/wazuh-indexer/jdk/bin/java -Xshare:auto -Dopensearch.networkaddress.cache.ttl=60 -Dopensearch.networkaddress.cache.negative.ttl=10 -XX:+AlwaysPreTouch -Xss1m -Djava.awt.headless=true -Dfile.encoding=UT>

Sep 11 15:40:39 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.OpenSearch (file:/usr/share/wazuh-indexer/lib/opensearch-2.13.0.jar)
Sep 11 15:40:39 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.OpenSearch
Sep 11 15:40:39 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: System::setSecurityManager will be removed in a future release
Sep 11 15:40:40 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: Sep 11, 2024 3:40:40 PM sun.util.locale.provider.LocaleProviderAdapter <clinit>
Sep 11 15:40:40 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: COMPAT locale provider will be removed in a future release
Sep 11 15:40:41 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: A terminally deprecated method in java.lang.System has been called
Sep 11 15:40:41 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.Security (file:/usr/share/wazuh-indexer/lib/opensearch-2.13.0.jar)
Sep 11 15:40:41 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: Please consider reporting this to the maintainers of org.opensearch.bootstrap.Security
Sep 11 15:40:41 ip-172-31-35-162.ec2.internal systemd-entrypoint[49637]: WARNING: System::setSecurityManager will be removed in a future release
Sep 11 15:41:01 ip-172-31-35-162.ec2.internal systemd[1]: Started wazuh-indexer.service - wazuh-indexer.
```

</details>

<details><summary>Wazuh Manager</summary>

```console
$ bash wazuh-install.sh -ws node_server
12/09/2024 10:20:04 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
12/09/2024 10:20:04 INFO: Verbose logging redirected to /var/log/wazuh-install.log
12/09/2024 10:20:05 INFO: Verifying that your system meets the recommended minimum hardware requirements.
12/09/2024 10:20:05 INFO: Wazuh repository added.
12/09/2024 10:20:06 INFO: --- Wazuh server ---
12/09/2024 10:20:06 INFO: Starting the Wazuh manager installation.
12/09/2024 10:21:25 INFO: Wazuh manager installation finished.
12/09/2024 10:21:25 INFO: Wazuh manager vulnerability detection configuration finished.
12/09/2024 10:21:25 INFO: Starting service wazuh-manager.
12/09/2024 10:21:46 INFO: wazuh-manager service started.
12/09/2024 10:21:46 INFO: Starting Filebeat installation.
12/09/2024 10:22:21 INFO: Filebeat installation finished.
12/09/2024 10:22:22 INFO: Filebeat post-install configuration finished.
12/09/2024 10:22:24 INFO: The filebeat.yml file has been updated to use the Filebeat Keystore username and password.
12/09/2024 10:22:52 INFO: Starting service filebeat.
12/09/2024 10:22:54 INFO: filebeat service started.
12/09/2024 10:22:54 INFO: Installation finished.

$ systemctl status wazuh-manager
● wazuh-manager.service - Wazuh manager
     Loaded: loaded (/usr/lib/systemd/system/wazuh-manager.service; enabled; preset: disabled)
     Active: active (running) since Thu 2024-09-12 10:22:49 UTC; 58s ago
      Tasks: 147 (limit: 9373)
     Memory: 6.0G
        CPU: 1min 17.936s
     CGroup: /system.slice/wazuh-manager.service
             ├─165109 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
             ├─165110 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
             ├─165113 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
             ├─165116 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
             ├─165160 /var/ossec/bin/wazuh-authd
             ├─165177 /var/ossec/bin/wazuh-db
             ├─165202 /var/ossec/bin/wazuh-execd
             ├─165218 /var/ossec/bin/wazuh-analysisd
             ├─165230 /var/ossec/bin/wazuh-syscheckd
             ├─165279 /var/ossec/bin/wazuh-remoted
             ├─165314 /var/ossec/bin/wazuh-logcollector
             ├─165335 /var/ossec/bin/wazuh-monitord
             └─165356 /var/ossec/bin/wazuh-modulesd

Sep 12 10:22:41 ip-172-31-47-54.ec2.internal env[165047]: Started wazuh-analysisd...
Sep 12 10:22:42 ip-172-31-47-54.ec2.internal env[165047]: Started wazuh-syscheckd...
Sep 12 10:22:43 ip-172-31-47-54.ec2.internal env[165047]: Started wazuh-remoted...
Sep 12 10:22:45 ip-172-31-47-54.ec2.internal env[165047]: Started wazuh-logcollector...
Sep 12 10:22:46 ip-172-31-47-54.ec2.internal env[165047]: Started wazuh-monitord...
Sep 12 10:22:46 ip-172-31-47-54.ec2.internal env[165352]: 2024/09/12 10:22:46 wazuh-modulesd:router: INFO: Loaded router module.
Sep 12 10:22:46 ip-172-31-47-54.ec2.internal env[165352]: 2024/09/12 10:22:46 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
Sep 12 10:22:47 ip-172-31-47-54.ec2.internal env[165047]: Started wazuh-modulesd...
Sep 12 10:22:49 ip-172-31-47-54.ec2.internal env[165047]: Completed.
Sep 12 10:22:49 ip-172-31-47-54.ec2.internal systemd[1]: Started wazuh-manager.service - Wazuh manager.

$ systemctl status filebeat
● filebeat.service - Filebeat sends log files to Logstash or directly to Elasticsearch.
     Loaded: loaded (/usr/lib/systemd/system/filebeat.service; enabled; preset: disabled)
     Active: active (running) since Thu 2024-09-12 10:22:24 UTC; 3min 25s ago
       Docs: https://www.elastic.co/products/beats/filebeat
   Main PID: 164814 (filebeat)
      Tasks: 8 (limit: 9373)
     Memory: 11.2M
        CPU: 179ms
     CGroup: /system.slice/filebeat.service
             └─164814 /usr/share/filebeat/bin/filebeat --environment systemd -c /etc/filebeat/filebeat.yml --path.home /usr/share/filebeat --path.config /etc/filebeat --path.data /var/lib/filebeat --path.logs /var/log/filebeat

Sep 12 10:22:24 ip-172-31-47-54.ec2.internal systemd[1]: Started filebeat.service - Filebeat sends log files to Logstash or directly to Elasticsearch..
```

</details>

<details><summary>Wazuh Dashboard</summary>

```console
$ bash wazuh-install.sh -fd -wd dashboard
11/09/2024 15:54:51 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 15:54:51 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 15:54:51 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 15:54:51 INFO: Wazuh web interface port will be 443.
11/09/2024 15:54:51 INFO: Wazuh repository added.
11/09/2024 15:54:52 INFO: --- Wazuh dashboard ----
11/09/2024 15:54:52 INFO: Starting Wazuh dashboard installation.
11/09/2024 15:56:55 INFO: Wazuh dashboard installation finished.
11/09/2024 15:56:55 INFO: Wazuh dashboard post-install configuration finished.
11/09/2024 15:56:55 INFO: Starting service wazuh-dashboard.
11/09/2024 15:56:56 INFO: wazuh-dashboard service started.
11/09/2024 15:57:59 INFO: Initializing Wazuh dashboard web application.
11/09/2024 16:00:00 WARNING: Cannot connect to Wazuh dashboard.
11/09/2024 16:00:25 WARNING: Failed to connect with node_indexer. Connection refused.
11/09/2024 16:00:25 INFO: --- Summary ---
11/09/2024 16:00:25 INFO: When Wazuh dashboard is able to connect to your Wazuh indexer cluster, you can access the web interface https://<wazuh-dashboard-ip>
    User: admin
    Password: ovHTJ6Ix?07k.+MLihz.Y1xHIy2xzb8p
11/09/2024 16:00:25 INFO: Installation finished.
```

</details>

### Installation without enough space

To perform these tests, different partitions were created to verify that the new changes are adapted correctly.

<details><summary>AIO installation</summary>

```console
$ df -h
Filesystem       Size  Used Avail Use% Mounted on
/dev/root        7.6G  3.6G  4.1G  47% /
tmpfs            3.9G     0  3.9G   0% /dev/shm
tmpfs            1.6G  852K  1.6G   1% /run
tmpfs            5.0M     0  5.0M   0% /run/lock
/dev/nvme0n1p15  105M  6.1M   99M   6% /boot/efi
tmpfs            787M  4.0K  787M   1% /run/user/1000
/dev/nvme0n1p2   1.8G  249M  1.5G  15% /usr/share

$ bash wazuh-install.sh -a
11/09/2024 14:40:46 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
11/09/2024 14:40:46 INFO: Verbose logging redirected to /var/log/wazuh-install.log
11/09/2024 14:40:57 INFO: Verifying that your system meets the recommended minimum hardware requirements.
11/09/2024 14:40:57 INFO: Wazuh web interface port will be 443.
11/09/2024 14:40:57 ERROR: There is not enough disk space in partition /dev/nvme0n1p2 to install AIO. Required: 2350MB, Available: 1479MB.
11/09/2024 14:40:57 INFO: --- Removing existing Wazuh installation ---
11/09/2024 14:40:57 INFO: Wazuh GPG key not found in the system
11/09/2024 14:40:57 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

</details>

<details><summary>Wazuh Indexer</summary>

```console
$ df -h
Filesystem        Size  Used Avail Use% Mounted on
devtmpfs          4.0M     0  4.0M   0% /dev
tmpfs             3.9G  976K  3.9G   1% /dev/shm
tmpfs             1.6G  448K  1.6G   1% /run
/dev/nvme0n1p1     30G  8.9G   22G  30% /
tmpfs             3.9G     0  3.9G   0% /tmp
/dev/nvme0n1p128   10M  1.3M  8.7M  13% /boot/efi
tmpfs             784M     0  784M   0% /run/user/1000
/dev/nvme0n1p2    827M  361M  408M  47% /usr/share

$ bash wazuh-install.sh -wi node_indexer
12/09/2024 13:37:11 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
12/09/2024 13:37:11 INFO: Verbose logging redirected to /var/log/wazuh-install.log
12/09/2024 13:37:11 INFO: Verifying that your system meets the recommended minimum hardware requirements.
12/09/2024 13:37:12 ERROR: There is not enough disk space in partition /dev/nvme0n1p2 to install wazuh-indexer. Required: 1050MB, Available: 478MB.
12/09/2024 13:37:12 INFO: --- Removing existing Wazuh installation ---
12/09/2024 13:37:12 INFO: Wazuh GPG key not found in the system
12/09/2024 13:37:12 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue. 
```

</details>

<details><summary>Wazuh Manager</summary>

```console
$ df -h
Filesystem        Size  Used Avail Use% Mounted on
devtmpfs          4.0M     0  4.0M   0% /dev
tmpfs             3.9G  896K  3.9G   1% /dev/shm
tmpfs             1.6G  460K  1.6G   1% /run
/dev/nvme0n1p1     30G  2.2G   28G   8% /
tmpfs             3.9G  4.0K  3.9G   1% /tmp
/dev/nvme0n1p128   10M  1.3M  8.7M  13% /boot/efi
tmpfs             784M     0  784M   0% /run/user/1000
/dev/nvme0n1p2    827M  290M  479M  38% /usr/share
/dev/nvme0n1p3    4.7G   24K  4.5G   1% /var

$ bash wazuh-install.sh -ws node_server
12/09/2024 13:48:36 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
12/09/2024 13:48:36 INFO: Verbose logging redirected to /var/log/wazuh-install.log
12/09/2024 13:48:42 INFO: Verifying that your system meets the recommended minimum hardware requirements.
12/09/2024 13:48:42 ERROR: There is not enough disk space in partition /dev/nvme0n1p3 to install wazuh-manager. Required: 7100MB, Available: 4533MB.
12/09/2024 13:48:42 INFO: --- Removing existing Wazuh installation ---
12/09/2024 13:48:42 INFO: Wazuh GPG key not found in the system
12/09/2024 13:48:46 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

</details>

<details><summary>Wazuh Dashboard</summary>

```console
$ df -h
Filesystem        Size  Used Avail Use% Mounted on
devtmpfs          4.0M     0  4.0M   0% /dev
tmpfs             3.9G  976K  3.9G   1% /dev/shm
tmpfs             1.6G  448K  1.6G   1% /run
/dev/nvme0n1p1     30G  8.9G   22G  30% /
tmpfs             3.9G     0  3.9G   0% /tmp
/dev/nvme0n1p128   10M  1.3M  8.7M  13% /boot/efi
tmpfs             784M     0  784M   0% /run/user/1000
/dev/nvme0n1p2    827M  361M  408M  47% /usr/share

$ bash wazuh-install.sh -fd -wd dashboard
12/09/2024 13:39:05 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
12/09/2024 13:39:05 INFO: Verbose logging redirected to /var/log/wazuh-install.log
12/09/2024 13:39:05 INFO: Verifying that your system meets the recommended minimum hardware requirements.
12/09/2024 13:39:05 INFO: Wazuh web interface port will be 443.
12/09/2024 13:39:05 ERROR: There is not enough disk space in partition /dev/nvme0n1p2 to install wazuh-dashboard. Required: 1130MB, Available: 478MB.
12/09/2024 13:39:05 INFO: --- Removing existing Wazuh installation ---
12/09/2024 13:39:05 INFO: Wazuh GPG key not found in the system
12/09/2024 13:39:06 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

</details>

**Comment by c-bordon** [comment]
For now we need the logic that allows us to manage the storage at the time of installation, in another instance we will review responsibilities of the amount of storage needed by each component, for now let's establish the storage that we already calculated taking into account an additional space just in case.

Once this is ready we create another issue to review the storage
